Add leaderboard reset button

rparrett opened the pull request:
It was a pretty amazing journey getting set up to develop and test this tiny changeset.

It's pretty basic. UI doesn't update when resetting.

Would be great to have this functionality when the bevy jam voting period begins. It's a bit of an ordeal to rebuild and re-upload to swap to a fresh leaderboard.

mockersf commented on jornet-server/src/domains/score.rs:
```diff
@@ -105,6 +112,7 @@ pub(crate) fn score() -> impl HttpServiceFactory {
         .wrap(cors)
         .route("{leaderboard_id}", web::post().to(save_score))
         .route("{leaderboard_id}", web::get().to(get_scores))
+        .route("{leaderboard_id}", web::delete().to(delete_scores))
```
putting the new endpoint here makes it reachable without authentication

You would need to put it in a scope that checks the bearer (for example https://github.com/vleue/jornet/blob/main/jornet-server/src/domains/leaderboard.rs#L58), and check that the leaderboard owner is the same as the token user

rparrett commented:
I moved it to `DELETE /leaderboards/{leaderboard_id}/scores` under the existing scope that validates tokens in `leaderboards.rs`.

I also added leaderboard ownership verification and tested both the Ok and Unauthorized responses.

mockersf commented:
looks good! do you want to move it out of draft?

mockersf commented:
deployed 👍
